src/plugins/ui_actions/public/service/ui_actions_execution_service.ts

```diff
@@ -0,0 +1,121 @@
+/*
```

Comment:
I think we should add unit tests for this service as there is some complicated async logic.